lib/datadog/core/telemetry/client.rb

```diff
@@ -12,26 +15,42 @@ class Client
         attr_reader \
           :emitter,
           :enabled,
+          :metrics_enabled,
           :unsupported,
-          :worker
+          :worker,
+          :metrics_worker
 
         include Core::Utils::Forking
 
         # @param enabled [Boolean] Determines whether telemetry events should be sent to the API
         # @param heartbeat_interval_seconds [Float] How frequently heartbeats will be reported, in seconds.
-        def initialize(heartbeat_interval_seconds:, enabled: true)
+        def initialize(heartbeat_interval_seconds:, metrics_enabled:, enabled: true)
           @enabled = enabled
+          @metrics_enabled = metrics_enabled
           @emitter = Emitter.new
           @stopped = false
           @unsupported = false
           @worker = Telemetry::Heartbeat.new(enabled: @enabled, heartbeat_interval_seconds: heartbeat_interval_seconds) do
             heartbeat!
           end
+
+          Metric::Rate.interval = heartbeat_interval_seconds
+
+          @metric_queue = MetricQueue.new
+
+          @metrics_worker = Telemetry::MetricWorker.new(
```

Comment:
Minor: The inconsistency between `metric` and `metrics` is a bit weird in these names